serving/using-external-dns.md

```diff
@@ -0,0 +1,171 @@
+
+## Prerequisite
+
+1. A Kubernetes cluster with [Knative Serving](https://github.com/knative/docs/blob/master/install/README.md) installed.
+1. Own the domain that will be used in Knative.
+1. Follow the steps [Edit using kubectl](https://github.com/knative/docs/blob/master/serving/using-a-custom-domain.md#edit-using-kubectl) to make Knative 
+use your owned domain.
+
+## Setup steps
+
+In this document, we use the GCP as an example to show how to set up 
+ExternalDNS. You can find detailed instructions for other cloud providers in [ExternalDNS documentation](https://github.com/kubernetes-incubator/external-dns#deploying-to-a-cluster).
+
+### Choose a DNS provider
+
+Skip this step if you already have a DNS provider for your owned domain.
+
+Here is a [list](https://github.com/kubernetes-incubator/external-dns#the-latest-release-v05) of DNS providers supported by ExternalDNS.
+Choose a DNS provider from the list.
+
+### Create a DNS zone for managing DNS records
+
+Skip this step if you already have a zone for managing DNS records of your 
+custom domain.
+
+A DNS zone which will contain the managed DNS records needs to be created.
+Assume your custom domain is `external-dns-test.my-org.do`. You can use the 
+following command to create a DNS zone if you are using Google Cloud DNS.
+```shell
+gcloud dns managed-zones create "external-dns-zone" \
+    --dns-name "external-dns-test.my-org.do." \
+    --description "Automatically managed zone by kubernetes.io/external-dns"
+```
+Make a note of the nameservers that were assigned to your new zone.
+```shell
+gcloud dns record-sets list \
+    --zone "external-dns-zone" \
+    --name "external-dns-test.my-org.do." \
+    --type NS
+```
+You should see the result like:
+```
+NAME                             TYPE  TTL    DATA
+external-dns-test.my-org.do.  NS    21600  ns-cloud-e1.googledomains.com.,ns-cloud-e2.googledomains.com.,ns-cloud-e3.googledomains.com.,ns-cloud-e4.googledomains.com.
+```
+In this case the DNS nameservers are `ns-cloud-{e1-e4}.googledomains.com`. But 
+yours could slightly differ, e.g. {a1-a4}, {b1-b4} etc.
+
+If this zone has the parent zone, you need to add NS records of this zone into 
+the parent zone so that this zone can be found from the parent.
+Assuming the parent zone is `my-org-do` and the parent domain is `my-org.do`, 
+and the parent zone is also hosted at Google Cloud DNS, you can follow below 
+steps to add the NS records of this zone into the parent zone. 
+```shell
+gcloud dns record-sets transaction start --zone "my-org-do"
+gcloud dns record-sets transaction add ns-cloud-e{1..4}.googledomains.com. \
+    --name "external-dns-test.my-org.do." --ttl 300 --type NS --zone "my-org-do"
+gcloud dns record-sets transaction execute --zone "my-org-do"
+```
+
+### Deploy ExternalDNS
+Apply the following manifest to install ExternalDNS. Note that you need to set 
+the argument `domain-filter` to your custom domain. 
+```
```

Comment:
can we check this in as a yaml file and link to it, so people can download?  alternatively if we have a link straight to external-dns's version that would be great.

Reply:
if we check this in, we could provide a one-liner to replace the `domain-filter` property and install

Reply:
similarly the one-liner could replace `provider=google` to something else, may be an env var.

Reply:
I chose to link straight to external-dns's version since we have to maintain our version to keep consistent with external-dns's version if we create our own one, which seems not very necessary.